utils/src/test/scala/com/salesforce/op/utils/stats/StreamingHistogramTest.scala

```diff
@@ -0,0 +1,250 @@
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package com.salesforce.op.utils.stats
+
+import breeze.integrate
+import breeze.stats.{meanAndVariance, MeanAndVariance}
+import breeze.stats.distributions._
+import com.salesforce.op.test.TestSparkContext
+import com.salesforce.op.utils.stats.StreamingHistogram.StreamingHistogramBuilder
+import com.salesforce.op.utils.stats.StreamingHistogramTest._
+import org.apache.spark.SparkConf
+import org.junit.runner.RunWith
+import org.scalatest.FlatSpec
+import org.scalatest.junit.JUnitRunner
+
+@RunWith(classOf[JUnitRunner])
+class StreamingHistogramTest extends FlatSpec with TestSparkContext {
+
+  val histogramSampleSize = 500
+  val mcSampleSize = 1000
+
+  // Enforce Kryo serialization check
+  conf.set("spark.kryo.registrationRequired", "true")
+
+  Spec(classOf[StreamingHistogram]) should "produce correct histogram distribution" in {
+    HistogramUtils.streamingHistogramBins(referenceHistogram).map {
+      case (point, count) => round(point) -> count
+    } should contain theSameElementsAs Seq(2.0 -> 1L, 9.5 -> 2L, 19.33 -> 3L, 32.67 -> 3L, 45 -> 1L)
+  }
+
+  it should "compute sum algorithm correctly" in {
+    val hist = referenceHistogram
+
+    round(hist.sum(0)) shouldEqual 0.0
+    round(hist.sum(2)) shouldEqual 0.5
+    round(hist.sum(9.5)) shouldEqual 2.0
+    round(hist.sum(15)) shouldEqual 3.28
+    round(hist.sum(20)) shouldEqual 4.65
+    round(hist.sum(35)) shouldEqual 8.03
+    round(hist.sum(45)) shouldEqual 10.0
+    round(hist.sum(46)) shouldEqual 10.0
+  }
+
+  it should "work with spark" in {
+    val data = sc.parallelize((0 to 10).map(_.toDouble), 2)
+    val histogram = {
+      val seqOp = (bldr: StreamingHistogramBuilder, point: Double) => {
+        bldr.update(point)
+        bldr
+      }
+      val combOp = (bldr1: StreamingHistogramBuilder, bldr2: StreamingHistogramBuilder) => {
+        bldr1.merge(bldr2.build)
+        bldr1
+      }
+
+      data.aggregate(new StreamingHistogramBuilder(15, 500, 1))(seqOp, combOp).build
+    }
+
+    HistogramUtils.streamingHistogramBins(histogram) should contain theSameElementsAs
+      (0 to 10).map(k => (k.toDouble, 1.0))
+  }
+
+  it should "correctly approximate standard normal distribution" in {
+    val sampleSize = 500
+    val gaussian1 = Gaussian(0, 1)
+    val gaussian2 = Gaussian(5, 5)
+    val mixture = MixtureDistribution(gaussian1, gaussian2, 0.8, sampleSize)
+
+    cdfTest("Gaussian(0, 1)", gaussian1, histogramSampleSize, 100, mcSampleSize, 50)
```

Comment:
let's not run every test 50 times. that is very slow. let's add this as val and maybe do just 3.